### ISSUE TITLE
Pattern Matching for "instanceof" operator should be used instead of simple "instanceof" + cast

### ISSUE THREAD

**Issue by nilsschmidt1337** [nilsschmidt1337/ldparteditor#865]
Structures like
```
int f(Object o) {
  if (o instanceof String) {  // Noncompliant
    String string = (String) o;
    return string.length();
  }
  return 0;
}
```
should be
```
int f(Object o) {
  if (o instanceof String string) {  // Compliant
    return string.length();
  }
  return 0;
}
```

**Comment by RishiKumarRay** [comment]
I want to raise PR for this? shall I ? also please let me know which are the folders where this is present

**Comment by nilsschmidt1337** [comment]
@RishiKumarRay sure, if you want to practice.
I can create an issue for you, too if you like.
Something, which is easy to do.

**Comment by RishiKumarRay** [comment]
@nilsschmidt1337 yes, please that would be a great help , Thanks

**Comment by nilsschmidt1337** [comment]
@RishiKumarRay thanks, I created issue #866 for you. Hope that it helps!

**Comment by nilsschmidt1337** [comment]
Closed with commit fdd578c.
